src/main/kotlin/org/rust/cargo/project/settings/ui/RustProjectSettingsPanel.kt

```diff
@@ -21,7 +21,10 @@ import java.nio.file.Path
 import java.nio.file.Paths
 import javax.swing.JLabel
 
-class RustProjectSettingsPanel(private val cargoProjectDir: Path = Paths.get(".")) : Disposable {
+class RustProjectSettingsPanel(
+    private val cargoProjectDir: Path = Paths.get("."),
+    private val updateListener: (() -> Unit)? = null
+) : Disposable {
```

Comment:
Do we really need to implement `Disposable` here?
We do nothing in `dispose` method

Reply:
We don't own any resources that need cleanup directly, but `UiDebouncer` does: it needs a parentDisposable in constructor.

Reply:
Hm, I think that perhaps instead `RustProjectSettingsPanel` can accept parent disposable as an argument instead?

Reply:
Oh, I've understood why we need `Disposable` here.

> Hm, I think that perhaps instead `RustProjectSettingsPanel` can accept parent disposable as an argument instead?

I think it would be not very convenient because we still use `RustProjectSettingsPanel` in `RustProjectConfigurable` directly and we don't have parent disposable.
Maybe don't change it at all?